### DISCUSSION TITLE
What is the maximum practical amount of vectors (dim 512) for FLAT index on one pc? What are the best alternatives with the highest recall rate?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#18203] [Q&A and General discussion] [answered]
What is the maximum practical amount of vectors (dim 512) for FLAT index on one pc? What are the best alternatives with the highest recall rate?
I mean I'm testing it right now using random data.

I have an accident an on ~ 2 millions of random entries (only first 100 are really entries) I had ran out of hard disk space, and after reboot I couldn't query even simplest querriesr like 

res = collection.query(
  expr = "pk == 1",
  output_fields = ["pk", "embeddings"],
)

**Comment by yhmo** [comment, marked as answer]
If you deployed milvus on a single machine, all the third-parties services are also on the machine, it will require a lot of disk space far beyond the 2 million entries. Because etcd, pulsar also have their data management, and log system continually writes log into the disk.
